Strictier regex to parse mypy errors

This regex is too loose and fails to parse mypy output when no column numbers are shown:
https://github.com/msherry/flycheck-pycheckers/blob/5119aa6cca41068988a36ae824e08d55613e0fdd/bin/pycheckers.py#L713

I would suggest to use instead `\d` to mean "digits" when line or column numbers are expected:
```
    output_matcher = re.compile(
        r'(?P<filename>[^:]+):'
        r'(?P<line_number>\d+):'
        r'((?P<column_number>\d+):)?'  # Column number is optional, depending on mypy options
        r' (?P<level>[^:]+):'
        r' (?P<description>.+)$')
```

Here is an example mypy line to show the problem:
```
filename.py:24: error: Dict entry 0 has incompatible type "str": "str"; expected "int": "str"
```

And a script to show the problem:
https://gist.github.com/lbolla/f81def50503998e6582368b59f98b03f

--- a/bin/pycheckers.py
+++ b/bin/pycheckers.py
@@ -709,8 +709,8 @@ class MyPy2Runner(LintRunner):
 
     output_matcher = re.compile(
         r'(?P<filename>[^:]+):'
-        r'(?P<line_number>[^:]+):'
-        r'((?P<column_number>[^:]+):)?'  # Column number is optional, depending on mypy options
+        r'(?P<line_number>\d+):'
+        r'((?P<column_number>\d+):)?'  # Column number is optional, depending on mypy options
         r' (?P<level>[^:]+):'
         r' (?P<description>.+)$')
 

